### ISSUE TITLE
XAmzContentSHA256Mismatch (2.4.x branch)

### ISSUE THREAD

**Issue by fooblahblah** [Kaliber/play-s3#54]
I noticed there was a play_2.4.x branch. I'm not sure what the status of that branch is, but when I try to use it with some existing code, based on the 5.0.2 release using Play 2.3, running under the play 2.4 6.0.0 build I get:

```
[info] A fully executed contract should
[error]   ! execute events in sequence
[error]    fly.play.s3.S3Exception: Problem accessing S3. Status 400, code XAmzContentSHA256Mismatch, message 'XAmzContentSHA256Mismatch'
[error]    Original xml:
[error]    Some(<Error><Code>XAmzContentSHA256Mismatch</Code><Message>The provided 'x-amz-content-sha256' header does not match what was computed.</Message><ClientComputedContentSHA256>598a590771d788600f9b7c6573ff1d32e0c18c54fc6c1350d3fff8596e51cf5d</ClientComputedContentSHA256><S3ComputedContentSHA256>876b5050d2b8cf5f7158144f1d8cdd20f1106a56be9735d7581843b4aab71f82</S3ComputedContentSHA256><RequestId>72793E0590C92FAF</RequestId><HostId>v0H7O7Kwp9fSNRhAJIt3SrCuV5td5CABMzpEpqYTXRxr4IoCf+70Yct0ASmAsN/tYlhvHXr7NKI=</HostId></Error>) (S3Exception.scala:15)
[error] fly.play.s3.S3Exception$.apply(S3Exception.scala:15)
[error] fly.play.s3.S3Response$.apply(S3Response.scala:9)
[error] fly.play.s3.Bucket$$anonfun$unitResponse$2.apply(Bucket.scala:260)
[error] fly.play.s3.Bucket$$anonfun$unitResponse$2.apply(Bucket.scala:260)
[error] akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
[error] akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
[error] akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
[error] akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
[error] akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
[error] akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
[error] akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
```

Any ideas? Sorry if this is a premature bug report.

**Comment by fooblahblah** [comment]
Ahh, I see this could be related to signer requirement now. Testing...

**Comment by fooblahblah** [comment]
Seems to still have an issue with signing the request

**Comment by EECOLOR** [comment]
As soon as I have some time to work on this I will.

The `play 2.4` branch is a development branch, so it's not complete. We will merge it into master once complete and create a `2.3.x` branch just before we do the merge.

**Comment by EECOLOR** [comment]
Related to #51

**Comment by EECOLOR** [comment]
A `2.4` version has been released, please check #51 and the README for details
